blockchain/state_manager/src/lib.rs

```diff
@@ -640,17 +639,14 @@ where
     ) -> Result<(Message, ApplyRet), Error> {
         // This isn't ideal to have, since the execution is syncronous, but this needs to be the
         // case because the state transition has to be in blocking thread to avoid starving executor
-        let outm: OnceCell<Message> = Default::default();
-        let outr: OnceCell<ApplyRet> = Default::default();
-        let m_clone = outm.clone();
-        let r_clone = outr.clone();
+        let (m_tx, m_rx) = std::sync::mpsc::channel();
+        let (r_tx, r_rx) = std::sync::mpsc::channel();
         let callback = move |cid: &Cid, unsigned: &ChainMessage, apply_ret: &ApplyRet| {
             if *cid == mcid {
-                let _ = m_clone.set(unsigned.message().clone());
```

Comment:
@lemmih I'm wondering whether this logic works before, setting cloned OnceCell does not seem to change the original one.

Check out this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8aa1aafe8a0b4936fa3b07fb61d49f0f)

```rust
use once_cell::sync::OnceCell;

fn main() {
    let cell = OnceCell::new();
    let cloned = cell.clone();
    cloned.set(1).unwrap();
    println!("cell: {:?}", cell.into_inner());
    println!("cloned: {:?}", cloned.into_inner());
}
```

```
cell: None
cloned: Some(1)
```

Reply:
Good catch! That was definitely broken.